internal/mode/static/state/graph/route_common.go

```diff
@@ -63,6 +69,27 @@ type RouteKey struct {
 	RouteType RouteType
 }
 
+type L4Route struct {
+	// ParentRefs describe the references to the parents in a Route.
+	ParentRefs []ParentRef
+	// Source is the source Gateway API object of the Route.
+	Source client.Object
+	// Spec is the L7RouteSpec of the Route
+	Spec L4RouteSpec
+	// Attachable indicates if the Route is attachable to any Listener.
+	Attachable bool
```

Comment:
I don't think this is being used anywhere yet. Let's leave it out until we actually need it.